Runtime/JavaScript/Tester/app.js

```diff
@@ -89,6 +93,68 @@ sm.registerState({
             cb(StateTokens.END, 51);
         }
     },
+    onTransition: function (data, cb) {
+        'use strict';
+        console.log(format('onNext({0}) [data: {1}]', this.token.name, data));
+        cb(StateTokens.TRANSITION);
+    },
+    onExit: function () {
+        'use strict';
+        console.log(format('onExit({0})', this.token.name));
+    }
+});
+
+sm.registerState({
+    token: StateTokens.TRANSITION,
+    counter: 0,
+    onInitialize: function () {
+        'use strict';
+        console.log(format('onInitialize({0})', this.token.name));
+        this.registerActionHandler(ActionTokens.NEXT, this.onNext);
+    },
+    onEnter: function (eventArg) {
+        'use strict';
+
+        var fromState = eventArg.from ? eventArg.from.name : '(null)';
+        var data = eventArg.data || '(null)';
+        console.log(format('onEnter({0}) [from: {1}, data: {2}]', this.token.name, fromState, data));
+
+        console.log('TRANSITION STATE:', (eventArg.triggeringActionToken || {}).name);
```

Comment:
Fallback `{}.name` is already known to be `undefined`.

```suggestion
        console.log('TRANSITION STATE:', eventArg.triggeringActionToken ? eventArg.triggeringActionToken.name : '(null)';
```